### PR TITLE
Implement button-base role menus (at least, adding them)+ some revamping of the project structure

### PR DESCRIPTION
**Pull request by VelvetToroyashi**

### CODE REVIEW

**Comment by VelvetToroyashi** [comment]
Have I mentioned that we need a CI pipeline so I can make sure these stupid merge conflicts don't break

**Comment by nullfact0r** [comment]
> Have I mentioned that we need a CI pipeline so I can make sure these stupid merge conflicts don't break

What do you mean exactly?

File: Dockerfile [hunk 1]
```diff
@@ -20,4 +20,4 @@ COPY --from=build /Silk/out .
 
 RUN chmod +x ./Silk.Core
 
-CMD ["./Silk.Core"]
+CMD ["./Silk.Core"]
```
Comment (nullfact0r) [inline Dockerfile:23]: 🤔 Github???

**Review by nullfact0r — APPROVED** [review]
LGTM

**Comment by VelvetToroyashi** [comment]
> > Have I mentioned that we need a CI pipeline so I can make sure these stupid merge conflicts don't break
> 
> What do you mean exactly?

Ask me 3 days ago

File: Dockerfile (hunk 1 above)
Comment (VelvetToroyashi) [reply Dockerfile:23]: Don't feel bad; I got a merge conflict on an entire file yesterday. A file created on the branch I was PRing onto master at that.

**Comment by nullfact0r** [comment]
I really love you sometimes, you know that?

**Comment by VelvetToroyashi** [comment]
Fixed merge conflicts.

**Pull request closed without merging**